motionplan/tpspace/ptgIKFrame.go

```diff
@@ -0,0 +1,60 @@
+package tpspace
+
+import (
+	"errors"
+	"math"
+
+	pb "go.viam.com/api/component/arm/v1"
+
+	"go.viam.com/rdk/referenceframe"
+)
+
+// ptgFrame wraps a tpspace.PrecomputePTG so that it fills the Frame interface and can be used by IK.
+type ptgIKFrame struct {
+	PrecomputePTG
+	limits []referenceframe.Limit
+}
+
+// NewPTGIKFrame will create a new frame intended to be passed to an Inverse Kinematics solver, allowing IK to solve for parameters
+// for the passed in PTG.
+func NewPTGIKFrame(ptg PrecomputePTG, dist float64) (referenceframe.Frame, error) {
+	pf := &ptgIKFrame{PrecomputePTG: ptg}
+
+	pf.limits = []referenceframe.Limit{
+		{Min: -math.Pi, Max: math.Pi},
+		{Min: -dist, Max: dist},
+	}
+	return pf, nil
+}
+
+func (pf *ptgIKFrame) DoF() []referenceframe.Limit {
+	return pf.limits
+}
+
+func (pf *ptgIKFrame) Name() string {
+	return ""
+}
+
+func (pf *ptgIKFrame) MarshalJSON() ([]byte, error) {
+	return nil, errors.New("marshal json not implemented for ptg IK frame")
+}
+
+func (pf *ptgIKFrame) InputFromProtobuf(jp *pb.JointPositions) []referenceframe.Input {
+	n := make([]referenceframe.Input, len(jp.Values))
+	for idx, d := range jp.Values {
+		n[idx] = referenceframe.Input{d}
+	}
+	return n
+}
+
+func (pf *ptgIKFrame) ProtobufFromInput(input []referenceframe.Input) *pb.JointPositions {
+	n := make([]float64, len(input))
+	for idx, a := range input {
+		n[idx] = a.Value
+	}
+	return &pb.JointPositions{Values: n}
+}
+
+func (pf *ptgIKFrame) Geometries(inputs []referenceframe.Input) (*referenceframe.GeometriesInFrame, error) {
+	return nil, errors.New("geometries not implemented for ptg IK frame")
```

Comment:
Shouldn't they be?

Reply:
Like why can't we just use the group frame to do this?

Reply:
The 3dof structure of the group frame does not play nicely with the IK solver, on account of the IK solver assuming a continuous distribution of float change -> position gradient, while the first Input to a group frame is an integer representing which PTG family to use. We want the 3dof frame for actually interacting with the base and forming the plan, but a 2dof frame for solving.